### PR TITLE
Add onnx, openvino, multi-gpu and removed memory leak

### PR DESCRIPTION
**Pull request by ryanchesler**
Added in onnx converted models as well as openvino runtime and removed memory leak introduced by multithread_exec

### CODE REVIEW

**Comment by maxjeblick** [comment]
I'm getting (on mr-0x remote RTX6000 machine)
```
ERROR: Could not find a version that satisfies the requirement openvino==2022.1.0 (from -r requirements-pt.txt (line 20)) (from versions: 2021.3.0, 2021.4.0, 2021.4.1, 2021.4.2)
ERROR: No matching distribution found for openvino==2022.1.0 (from -r requirements-pt.txt (line 20))
```

**Comment by maxjeblick** [comment]
Thanks for this great work! I checked inference speed on `AMD Ryzen Threadripper PRO 3975WX 32-Cores`; inference time for `db_resnet50_onnx` is approx. 1.5-1.9 times faster than `db_resnet50`.


Some general things I noticed:
- I had some issues with installing the new requirements. Pip's new dependency manager failed to resolve some conflicts. 
- Several tests fail, as openvino is not found when imported.
- On RTX6000, the default OCR predictor does not work, as the default torch version is not compatible. This is also true for the current master branch. There, `ocr_predictor` does not automatically switch to GPUs. Having an additional flag in `ocr_predictor` may be useful (e.g. `use_gpu`).
- I would suggest to subclass from DBNet (or to copy relevant code parts over) rather than modifying DBNet code itself. The latter may break some workflows/parts of the repository.

File: doctr/models/predictor/pytorch.py [hunk 1]
```diff
@@ -16,6 +16,8 @@
 from doctr.utils.geometry import rotate_boxes, rotate_image
 
 from .base import _OCRPredictor
+import pdb
```
Comment (maxjeblick) [inline doctr/models/predictor/pytorch.py:19]: This is probably unused?

Comment (ryanchesler) [reply doctr/models/predictor/pytorch.py:19]: Yes, I will remove this.

**Comment by ryanchesler** [comment]
Working on fixing the dependencies and making sure it works on cpu and gpu automatically. I put the requirements in the wrong part initially. Now trying to figure out how to get openvino to work. It has a strict constraint on its numpy version that makes it incompatible on paper with everything else. I know it works on much newer numpy versions though. Just need to make it ignore openvinos claimed dependency somehow.

**Comment by cooleel** [comment]
@ryanchesler do you have an idea why half of the tests failed on 
```
ERROR: No matching distribution found for numpy<2.0.0,>=1.23.0
```

**Comment by ryanchesler** [comment]
I added return_model_outputs option into predictor. This simplified the output for the rotation check model and makes it more robust so it takes in all of my doctr changes and can use onnx, or pytorch, half precision, gpu, cpu, etc. Whatever is available more seamlessly. Uses the same switching logic. https://github.com/h2oai/argus-ocr/blob/ocr_cleanup/argus/ocr.py#L771

**Comment by cooleel** [comment]
These failed tests are expected, from styling, recognition training and disabled pdf functions by us.

**Pull request merged**